app/utils/function.ts

```diff
@@ -44,6 +44,27 @@ export function throttle<T extends (...args: any) => any>(
   }
 }
 
+export function block<T extends (...args: any) => any>(
+  fn: T
+): ThrottledFunction<T> {
+  let executing: boolean
+  let lastResult: ReturnType<T>
+  
+  return async function (this: any, ...args: any[]): Promise<ReturnType<T>> {
+    const context = this
+    
+    if (!executing) {
```

Comment:
I don't know what this function is supposed to do but i'm pretty sure it's bugged, since calling this function twice in a row will set executing to false no matter the timing of the asynchronous task

Reply:
Good catch

Reply:
You may want to add a comment to clarify what the function does. It's a bit strange to me because usually if i want an async function to be called once and return the appropriate result, i store the promise, not the lastResult